File: src/main/java/xyz/aeolia/ashutils/task/UserPruneTask.java
package xyz.aeolia.ashutils.task;

import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;
import xyz.aeolia.ashutils.user.UserHelper;

public class UserPruneTask extends BukkitRunnable {
  Player player;

  public UserPruneTask(Player player) {
    this.player = player;
  }

  @Override
  public void run() {
    if (player.isOnline()) return;
    UserHelper.removeUser(player.getUniqueId());
    Bukkit.getLogger().info("[AshUtils] Pruned user " + player.getName() + ".");
  }
}

File: src/main/java/xyz/aeolia/ashutils/task/UserPruneTask.kt
package xyz.aeolia.ashutils.task

import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.scheduler.BukkitRunnable
import xyz.aeolia.ashutils.manager.UserManager

class UserPruneTask(val player: Player) : BukkitRunnable() {
  override fun run() {
    if (UserManager.getUser(player).online) return
    UserManager.removeUser(player)
    Bukkit.getLogger().info("[AshUtils] Pruned user ${player.name}")
  }
}